### PR TITLE
Update simple smart answers

### PR DESCRIPTION
**Pull request by andysellick**
## What

Updates simple smart answers to use govuk-frontend markup for the page and also to use components properly - radio component was in use but error messages, hints and question titles were still being rendered external to the component.

## Why

Removes unnecessary CSS and markup and improves component use. This came about from looking into what sass in `static` was needed and what wasn't.

## Visual changes

- Example simple smart answer: https://www.gov.uk/settle-in-the-uk
- Test URL: https://govuk-fronte-update-sim-kcpuvt.herokuapp.com/settle-in-the-uk

---

**Before:**

Screenshot 2020-01-23 at 12 55 13

Screenshot 2020-01-23 at 12 56 45

---

**After:**

Screenshot 2020-01-23 at 12 55 21

Screenshot 2020-01-23 at 12 56 56

### CODE REVIEW

File: app/views/simple_smart_answers/_completed_question.html.erb [hunk 1]
```diff
@@ -1,10 +1,11 @@
-<li class="done">
-  <h3 class="question">
-    <span class="question-number"><%= completed_question_counter + 1 %></span><%= completed_question.question.title %>
```
Comment (injms) [inline app/views/simple_smart_answers/_completed_question.html.erb]: Would there removal of headings make it less scannable for people using screen readers to pull out the headings in the page?

File: app/views/simple_smart_answers/_completed_question.html.erb [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-  </h3>
-
-  <div class="answer"><%= completed_question.label %></div>
-  <p class="undo">
-    <%= link_to "Change this answer", change_completed_question_path(completed_question_counter), class: "govuk-link" %>
-  </p>
-</li>
+<div class="govuk-summary-list__row">
```
Comment (injms) [inline app/views/simple_smart_answers/_completed_question.html.erb]: This description list feels as if it's trying to be a table - so I wonder if a table would be a more semantic way of displaying this information?

File: app/views/simple_smart_answers/_outcome.html.erb [hunk 3]
```diff
@@ -1,13 +1,10 @@
-<div class="article-container">
-  <div class="content-block outcome group" data-module="track-smart-answer" data-smart-answer-node-type="outcome" data-smart-answer-slug="<%= outcome.slug %>">
-    <div class="inner group">
-      <div class="result-info">
-        <div class="summary"><h2 class="result-title"><%= outcome.title %></h2></div>
+<div data-module="track-smart-answer" data-smart-answer-node-type="outcome" data-smart-answer-slug="<%= outcome.slug %>">
+  <%= render "govuk_publishing_components/components/heading", {
```
Comment (injms) [inline app/views/simple_smart_answers/_outcome.html.erb]: Components ftw :taco:

File: app/views/simple_smart_answers/flow.html.erb [hunk 4]
```diff
@@ -3,19 +3,15 @@
...
-      <div class="start-again">
-        <%= link_to "Start again", simple_smart_answer_path(@publication.slug, :edition => @edition), class: "govuk-link" %>
-      </div>
-      <ol>
-        <%= render :partial => "completed_question", :collection => @flow_state.completed_questions %>
-      </ol>
+    <div class="start-over">
+      <%= link_to "Start again", simple_smart_answer_path(@publication.slug, :edition => @edition), class: "govuk-link" %>
     </div>
+    <dl class="govuk-summary-list govuk-!-margin-bottom-9">
```
Comment (injms) [inline app/views/simple_smart_answers/flow.html.erb]: I thought that we (as in, we the GOV.UK Frontenders) decided that the `!important` override classes shouldn't be used if possible?

File: app/views/simple_smart_answers/_outcome.html.erb (hunk 3 above)
Comment (injms) [reply app/views/simple_smart_answers/_outcome.html.erb]: Also, autocorrect ftw.

File: app/views/simple_smart_answers/_completed_question.html.erb (hunk 1 above)
Comment (andysellick) [reply app/views/simple_smart_answers/_completed_question.html.erb]: That's a good point, but I don't think that every question in this bit needs to be a heading. I've added some visually hidden headings for 'your answers' and 'current question', see new commit.

File: app/views/simple_smart_answers/_completed_question.html.erb (hunk 2 above)
Comment (andysellick) [reply app/views/simple_smart_answers/_completed_question.html.erb]: That's what I thought too - in fact when I rebuilt `smart-answers` I did the equivalent section using a table. I was going to do the same here but then I found [some Design System guidance](https://design-system.service.gov.uk/patterns/check-answers/) on the subject.

Using a description list is also a bit more responsive - see [a current smart-answer](https://www.gov.uk/maternity-paternity-calculator/y/paternity) for the table use.

File: app/views/simple_smart_answers/flow.html.erb (hunk 4 above)
Comment (andysellick) [reply app/views/simple_smart_answers/flow.html.erb]: Ack, sorry, missed that. Will correct.

File: app/views/simple_smart_answers/_completed_question.html.erb (hunk 2 above)
Comment (injms) [reply app/views/simple_smart_answers/_completed_question.html.erb]: Super - in that case I'm happy with the use of description list.

**Pull request merged**